File: Garnik/CS50/Week1/credit.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <math.h>

int main(void)
{
    long credit = get_long("Enter the credit number: ");
    char str[17];
    //convert long to string
    sprintf(str, "%li", credit);
    int length = strlen(str);
    bool x = true;
    int sum2 = 0;

    while (credit > 0)
    {
        long temp;
        //get every last digit
        long tiv = credit % 10;

        if (!x)
        {
            tiv = 2 * tiv;
            temp = tiv / 10 + tiv % 10;
            sum2 = sum2 + temp;
            x = true;
        }
        else
        {
            sum2 = sum2 + tiv;
            x = false;
        }
        credit = credit / 10;
    }

    if (sum2 % 10 == 0)
    {
        if (length == 15 && str[0] == '3' && (str[1] == '4' || str[1] == '7'))
        {
            printf("AMEX\n");
        }
        else if ((length == 13 || length == 16) && (str[0] == '4' || str[0] == '2'))
        {
            printf("VISA\n");
        }
        else if (length == 16 && str[0] == '5' &&  str[1] <= '5')
        {
            printf("MASTERCARD\n");
        }
        else
        {
            printf("INVALID\n");
        }

    }
    else
    {
        printf("INVALID\n");
    }
}

File: Garnik/CS50/Week6/Credit/credit.py
from cs50 import get_string, get_int

credit = (str(get_int("Please enter the CC number: ")))[::-1]

even = 0
for ev in credit[0::2]:
    even = even + int(ev)

odd = 0
for val in credit[1::2]:
    odd = odd + ((int(val) * 2)//10 + (int(val) * 2) % 10)

cc = odd + even
if cc % 10 == 0:
    if len(credit) in [13, 15, 16]:
        credit = int(credit)
        if credit % 10 == 4 or credit % 10 == 2:
            print("VISA")
        elif credit % 100 == 73 or credit % 100 == 43:
            print("AMEX")
        elif credit % 100 >= 15 or credit % 100 <= 55:
            print("MASTERCARD")
        else:
            print("INVALID")
        exit(0)

print("INVALID")
